addons/mail/models/res_partner.py

```diff
@@ -81,9 +80,16 @@ def _notify_prepare_template_context(self, message):
         if message.res_id and message.model in self.env:
             record = self.env[message.model].browse(message.res_id)
 
+        if record and hasattr(record, 'company_id'):
+            company_id = record.company_id.id
+        else:
+            company_id = user.company_id.id
+        company_name = user.company_id.name
```

Comment:
Hum, there might be an inconsistency between company_id and company_name, then.
Could be:
        if record and hasattr(record, 'company_id'):
            company = record.company_id
        else:
            company = user.company_id
        company_id = company.id
        company_name = company.name